### DISCUSSION TITLE
Amortisations-Rechner

### DISCUSSION THREAD

**Discussion by Zeppelin-PV** [solectrus/solectrus#2105] [General]
könnte mir einen Amortisations-Rechner in Solectrus gut vorstellen.
Vielleicht auf der Ebene TopTen, Einstellungen
Das Modul würde nach meiner Vorstellung anzeigen wie weit sich die Anlage schon amortisert hat. Anzeige Restsumme ...
Inbetriebnahmedatum, Kaufpreis müssten eingegeben werden.
Solectrus rechnet immer den aktuellen finanziellen Stand. 
Auch eine Prognose wäre super, die auf Grund der bisherigen Daten anzeigt in wann etwa die Anlage auf 0 Euro kommt.
Je länger die Betriebszeit je genauer die Prognose.

**Comment by holiday-sunrise** [comment]
Würdest du die Werte in der .env Voreinstellen ? Kaufpreis

**Comment by Zeppelin-PV** [comment]
sorry, wenn es um das 'Wie' geht bin ich raus !
Ich bin absoluter Laie was das programmieren angeht. Ich habe es zwar geschafft Solectrus auf einen Raspi zu bringen, aber das war's dann schon.
Begeistert bin ich allemal, deshalb fallen mir auch immer so Kleinigkeiten ein.
Befürworten würde ich alles was für den Benutzer einzugeben wäre, quasi ein User Interface mit tollen Möglichkeiten; siehe dazu meine Sammlung in #1874
... aber so ein Amortisationsrechner wäre schon toll, oder ?

**Comment by StevieC121176** [comment]
Klingt spannend. Ich könnte mir aber vorstellen das das erheblichen Aufwand mit sich bringt und die Rechnung recht kompliziert sein könnte. Es gibt doch etliche Faktoren von denen es abhängt.

**Reply by halmich** [reply]
Ja, es gibt etliche Faktoren. Jedoch reicht bei der großen Mehrheit wohl der Anschaffungspreis und die von Solectrus errechnete Ersparnis.
Beispiel:
Anschaffung 20.000 Euro netto
Ersparnis im Jahr: 2000 Euro
Ammortisierung in 10 Jahren.
Kleinere Kulanzzahlungen wirken sich fast nicht aus, Versicherungen haben nur sehr wenige Betreiber. Einzig vielleicht größere Reparaturen verändern die Ammortisierung wesentlich...
Für die Berechnung in Solectrus wäre folgendes denbar:
Durchschnittl. Ersparnis = Ersparnis : Anzahl der Tage * 365.    
Kosten der anlage / durchschnittl. Ersparnis
Der Wert wird nach 2 Jahren auch nicht mehr wesentlich schwanken.

**Reply by ledermann** [reply]
Ein nicht zu vernachlässigender Faktor ist die Strompreis-Entwicklung, die in SOLECTRUS ja bereits erfasst wird. Durch den höheren Strompreis im Laufe der Jahre hat sich die Ersparnis deutlich erhöht. Waren es im ersten Jahr noch [€1724](https://solectrus.ledermann.dev/inverter_power/2021), sind es im laufenden Jahr bereits [€2261](https://solectrus.ledermann.dev/inverter_power/2023) - und das Jahr ist noch lange nicht zu Ende. 

Diese Schwankungen möchte ich gerne mit berücksichtigen. Im Prinzip ist klar, wie die Berechnung erfolgen muss, allein es fehlt mir momentan die Zeit für die Implementierung :)

**Reply by halmich** [reply]
@ledermann Ja, aber dies wird in deiner bisherigen Berechnung Ersparnis ja berücksichtigt, und auch in meiner Formel oben für die Ammortisierung.
Gesamt Ersparnis (seit IBN) : Anzahl der Tage (seit IBN) * 365 ergibt die Ersparnis pro Jahr, tagesgenau gemittelt.
Die Kosten durch diese Ersparnis sollte eine sehr gute Hochrechnung sein.
Meine Werte: 
Ersparnis lt. Solectrus: 6185
Tage seit Inbetriebnahme: 844
Ersparnis pro Tag: 7,328€ , damit pro Jahr: 2674,79€
Anschaffung: 19.000 : 2674,79 = 7,1 Jahre.
Inbetriebnahme: 11.06.2021  
Ammortisiert am ca. 15.08.2028
Laufzeit bis 31.12.2041:  (also noch 6664 Tage)
Ergibt momentan prognostizierte Gesamtersparnis von ca. 55000 Euro, abzgl. Kosten von 19.000 Euro macht dies 36.000 Euro Gewinn.

Was ich sagen will - die bisherigen Strompreisschwankungen hast du schon mit drin! und wie sich der Preis künftig entwickeln wird, weiß keiner. Das kriegst du nicht rein (bzw. nur mit verschiedenen Modellen..., lohnt sich m.E. nicht).

**Reply by ledermann** [reply]
Ok, jetzt habe ich verstanden, was du meinst. Ja, volle Zustimmung, genauso könnte man es machen. Ich habe schon mal ein Issue angelegt: https://github.com/solectrus/solectrus/issues/2416

**Comment by Zeppelin-PV** [comment]
in der Einsparungsberechnung von Solectrus steckt doch eigentlich das Wesentliche
![grafik](https://github.com/solectrus/solectrus/assets/126392805/bf4708f9-b119-4e0c-871e-1dad350153db)
Der Kaufpreis minimiert sich aus dieser Berechnung
... oder mache ich da einen Denkfehler

**Comment by ledermann** [comment]
Zunächst einmal finde ich den Vorschlag sehr gut, @Zeppelin-PV! Ich hatte da auch schon mal dran gedacht, aber wegen des Aufwands erst mal nicht weiter verfolgt. Wie @StevieC121176 schon sagt, gibt es weitere Faktoren. Ich sehe beispielsweise folgende Ausgaben bzw. Einnahmen, die man vermutlich mit einberechnen möchte:

- SENEC-Kulanzleistungen ;-)
- Versicherungsprämien
- Nachträgliche Aufrüstung der Anlage (weitere Module, weiteres Dach, Balkonkraftwerk etc.)
- Kosten für Reparaturen und Wartung

Denkbar wäre das natürlich. Jeder dieser Posten wäre mit einem Datum und einem Betrag zu versehen. Der Rest ist einfache Finanzmathematik. Man könnte jederzeit nachsehen, zu wieviel Prozent sich die Anlage bereits amortisiert hat. Und man könnte prognostizieren, wann die 100% erreicht sein werden.

In der `.env`-Datei würde ich die Input-Daten nicht speichern wollen. Das gehört meiner Ansicht nach in die Datenbank (wo bereits die Strompreise stehen), wo es dann über das UI zu editieren wäre. 
Die `.env`-Datei war ganz zu Anfang aus Nerd-Sicht ein einfacher Konfigurationsweg, aus Sicht eines Normalsterblichen ist es eher unschön. Langfristig möchte ich in der `.env-Datei` nur Passwörter, Token, IP-Adressen und anderes technisches Zeug ablegen. Alles andere soll in die Datenbank.

**Comment by StevieC121176** [comment]
Das würde voraussetzen das die Anlage zur gleichen Zeit mit Solectrus in Betrieb geht, wie berechnet man sonst die Amortisierung für die Differenz zwischen Inbetriebnahme Solaranlage und Installation Solectrus (In meinem Fall ging erst die Anlage in Betrieb, etwas später die Batterie und Solectrus hab ich erst seit 3 Wochen im Betrieb (wohl gemerkt läuft wie ein Uhrwerk)
BtW viele Firmen haben auch ein schlechtes bzw. gar kein Monitoring, da wird es noch schwieriger)

**Reply by ledermann** [reply]
Damit die Amortisationsberechnung überhaupt funktionieren kann, müssen natürlich vollständige Daten vorliegen (Erzeugung, Verbrauch und Einspeisung seit Inbetriebnahme der Anlage). Bei SENEC ist das glücklicherweise immer der Fall, weil man [historische Daten importieren](https://github.com/solectrus/senec-importer) kann. 

Wenn man etwas anderes als SENEC nutzt, fehlen diese Daten und die Berechnung wird verzerrt - je nachdem, wie viel Zeit zwischen Anlagen-Inbetriebnahme und SOLECTRUS-Installation vergangen ist. Vielleicht liefern die anderen Hersteller auch historische Daten, die man importieren könnte. 

Da aber vermutlich der Großteil der SOLECTRUS-User aus dem SENEC-Lager kommt, macht das Feature meines Erachtens durchaus Sinn.

**Reply by StevieC121176** [reply]
Ich finde das Feature auch klasse. Zu den Daten, also Kostal liefert auch historische Daten die man als csv aus den WR importieren kann. Man kann in den WR auch Die Gesamteinspeisung , Gesamtverbrauch und Gesamterzeugung seit Inbetriebnahme ablesen. Das sollte bei anderen WR auch möglich sein. Vielleicht könnte man die benötigten Werte einmal als Startwerte eingeben, welche Solectrus dann zur Berechnung heranziehen könnte.

**Comment by ledermann** [comment]
Interessant, somit sehe ich zwei verschiedene Dinge, die man diskutieren könnte:

1. Manuelle Erfassung von Gesamteinspeisung, Gesamtverbrauch und Gesamterzeugung bis zum Zeitpunkt X, an dem SOLECTRUS "übernommen" hat. Damit wäre dann eine Amortisationsrechnung möglich, auch wenn SOLECTRUS nicht von Anfang an gemessen hat. Na ja, nicht ganz: Da es in der Vergangenheit höchstwahrscheinlich verschiedene Strompreise gegeben hat, können die Gesamtwerte (in kWh) nur grobe Anhaltswerte sein. Insofern wäre der folgende Ansatz besser.

2. Neues Tool namens "Kostas-Importer", das analog zum [SENEC-Importer](https://github.com/solectrus/senec-importer) historische Daten einliest und somit in SOLECTRUS verfügbar macht. Das wäre unabhängig vom Amortisationsrechner sinnvoll, z.B. für die Top10 und die bereits jetzt mögliche Ersparnis- und Kosten-Berechnung. @StevieC121176: Wenn Du Interesse hast, kannst Du mir ja mal eine Beispiel-CSV zuschicken. Wenn das ähnlich aufgebaut ist wie bei SENEC, dürfte ein Import-Tool ohne großen Aufwand möglich sein.

**Comment by StevieC121176** [comment]
Gerne, hier mal eine Beispiel CSV mit 2 Tagen.

[log-7.csv](https://github.com/solectrus/solectrus/files/11816458/log-7.csv)

Zu 2. sehe ich bei Kostal ein Problem wenn man wie in meinem Fall 2 oder mehr WR schwarmverschaltet. Es werden in den CSV nur die Daten des jeweiligen WR gespeichert, keine Gesamtdaten aller verschalteter WR. Das passiert im Kostal Smartmeter, welches natürlich keinen CSV Export hat bzw im Kostal Solar Portal. 

Daher denke ich wäre 1. die bessere Alternative.

**Reply by StevieC121176** [reply]
Aus dem Solarportal kann man sich Wochenweise CSV Daten herunterladen:

[chart-2.csv](https://github.com/solectrus/solectrus/files/11816545/chart-2.csv)

**Reply by StevieC121176** [reply]
Ich habe mir nochmal das Kostal Solar Portal näher angeschaut. Also da kann man auch bei komplexen Anlagen mit mehreren WR und Batterien CSV Dateien mit Tages, Wochen, Monats und Jahresübersichten herunterladen, in denen schon alle WR zusammengerechnet sind. Dann würde ein Kostal Importer wahrscheinlich doch funktionieren. Leider kenne ich die vom Senec importer CSV Dateien nicht.

**Reply by ledermann** [reply]
Klingt alles gut. Ich habe mal eine neue Diskussion für die Import-Idee angelegt und antworte dort, damit wir hier beim Thema "Amortisation" bleiben können:
https://github.com/orgs/solectrus/discussions/2110

**Comment by ThePhanThomas** [comment]
Hallo zusammen,

zum Thema Amortisationsrechnung: Mir fiel im Solectrus, aber auch in der Senec-App schon auf, dass die Werte für verbrauchten Strom gerechnet aufs Jahr deutlich höher sind als vor Installation der PV-Anlage. Wir hatten bspw. vor Montage einen Verbrauch von ca. 2400 bis 2500kWh im Jahr. 

Im März wurde die Anlage installiert. Wenn ich für April bis Juni mal den gemessenen Verbrauch summiere, komme ich auf 853kWh. Im Winter verbraucht man ja eher mehr Strom, aber ignoriere ich das mal und nehme an, alle Quartale seien gleich, käme ich Stand jetzt auf über 3400kWh. 

Ich habe gelesen, dass Senec in seine Verbrauchsdaten die Verluste des Wechselrichters mit einrechnet (quasi als Hausverbrauch – ca. 50 Watt je "produzierter" Kilowattstunde). Aber selbst damit haut's nicht hin. Wie dem auch sei, der Stromverbrauch ohne Anlage wäre deutlich niedriger als mit der Anlage – zumindest anhand der gemessenen Werte. Ich denke mal, dass nicht nur ich das beobachte?

Für eine "echte" Amortisationsrechnung müsste das irgendwie mit berücksichtigt werden, oder?

**Comment by ledermann** [comment]
Danke für den Hinweis, @ThePhanThomas. Ja, der Eigenverbrauch ist wohl erheblich, lässt sich aber nicht genau beziffern. Nach meinem Kenntnisstand gibt es von SENEC auch keine Angaben dazu, jedenfalls kenne ich keine. Richtig ist, dass es im Hausverbrauch enthalten sind. Aktuell gibt es auf Facebook [eine Diskussion dazu](https://www.facebook.com/groups/4444909445612640/permalink/5758787104224861/
), wo dieser Link geteilt wurde:
https://www.ikz.de/detail/news/detail/wie-viel-strom-zieht-mein-system/

Für SOLECTRUS heißt das: Eigentlich müsste dieser Eigenverbrauch nicht nur bei der (zukünftig zu realisierenden) Amortisationsrechnung berücksichtigt werden, sondern auch bei der (bereits implementierten) Berechnung der Einsparung.

Fragt sich nur, wie man das angeht. Wenn man unterstellt, dass der Eigenverbrauch proportional zur Erzeugung ist, dann könnte man mit einem Korrekturfaktor arbeiten, vielleicht in der Größenordnung von 10%. Dieser Faktor wird dann sowohl bei der Einsparungsberechnung, als auch bei der Amortisationsrechnung verwendet. Der User sollte den Faktor anpassen können. 

Ich würde das aber nicht zu weit treiben, also z.B. nicht verschiedene Faktoren einführen (einen für Wechselrichter, einen für Wandlungsverluste durch den Akku usw.).

**Reply by ThePhanThomas** [reply]
Genau, ein Korrekturfaktor würde wahrscheinlich helfen. Den kann ich aber natürlich auch gedanklich ansetzen (mach ich auch). Wenn ich die Verbräuche der letzten Jahre nehme und weiß, da gab es keine größeren Anschaffungen, dann kann die ja mitteln und hinterher als Maßstab bei der Berechnung der tatsächlichen Einsparung ansetzen. So zumindest der Plan. Ob und wie sich das sinnvoll in die App einbauen ließe, puh, schwierig, weil's am Ende ja dann nicht auf konkreten Werten basiert.

**Comment by Zeppelin-PV** [comment]
ich habe seit gut einem Monat auf der dritten, von Senec unbenutzen Phase ein BKW am laufen.
Der Enfluri registriet den Strom und weist ihn auch mit einem Minus aus bei Einspeisung.
Wenn auf allen drei Phasen eingespeißt wird und quasi der Hausverbrauch Null ist dann müsste der angezeigte Wert unter Hausverbrauch
den Verbrauch des Senec System darstellen; oder hab ich da einen Gedankenfehler ?

![grafik](https://github.com/solectrus/solectrus/assets/126392805/0be75b3b-09de-48d2-8bbe-847eb4455a97)

**Comment by Zeppelin-PV** [comment]
... das ist ja aber nur eine Momentaufnahme
sobald L3 wieder Plus Werte anzeigt weiß man schon nicht mehr ist es das Haus oder der Senec.
Meine täglichen Einträge in eine Excel Liste zeigen aber sehr genau, dass im Winter das Haus zwar mehr verbraucht aber der Verbrauch kleiner ist als z.B. diesen Juni. Dies kann man schon eindeutig dem Senec Verbrauch zuschieben.
Habe auch versucht über die Liste eine gewisse Abhängigkeit abzuleiten, 
Komm aber auf keinen grünen Zweig, zumindest nicht was man für eine Amortisationsrechnung benutzen könnten.

**Comment by mzyc64** [comment]
Die Amortisationsrechnung mit manueller Zuerfassung von sonstigen Erlösen und Aufwendungen würde ich auch hilfreich finden. Gibt es hierzu bereist weiter Pläne?

**Reply by ledermann** [reply]
Außer dem Plan, ein solches Feature zukünftig zu realisieren, gibt es noch keine konkreteren Überlegungen. 

Das ist ein wenig das Schicksal einer Open-Source-Anwendung, die von einem Entwickler alleine in der Freizeit betrieben wird. Je nach verfügbarer Zeit arbeite ich daran und bevorzuge dabei Features, die auch mir persönlich einen Mehrwert bringen. Einen Amortisationsrechner finde ich aber auch selbst interessant, insofern könnte das was werden :-)

**Comment by ledermann** [comment]
Der Vollständigkeit halber hier der Hinweis, dass die Amortisationsberechnung zwischenzeitlich implementiert wurde und in das nächste Update (v1.3) einfließen wird:
https://github.com/solectrus/solectrus/issues/2416
